### ISSUE TITLE
Defining custom field types

### ISSUE THREAD

**Issue by tmc** [facebook/ent#1097]
github.com/facebook/ent/schema/field.Descriptor doesn't export the err field so this means that authors of custom types (one based on the uuid field here) can't set this field to indicate an error.

Should err be exported so custom types can be authored outside of the ent field package?

```
func (b *pluidBuilder) Default(fn interface{}) *pluidBuilder {
	typ := reflect.TypeOf(fn)
	if typ.Kind() != reflect.Func || typ.NumIn() != 0 || typ.NumOut() != 1 || typ.Out(0).String() != b.desc.Info.String() {
		b.desc.err = fmt.Errorf("expect type (func() %s) for pluid default value", b.desc.Info)
	}
	b.desc.Default = fn
	return b
}
```

**Comment by a8m** [comment]
Hey @tmc! 

> Should err be exported so custom types can be authored outside of the ent field package?

Sure, you’re welcome to send a patch for this, or I can change it later today. Thanks!

**Comment by tmc** [comment]
> Hey @tmc!
> 
> > Should err be exported so custom types can be authored outside of the ent field package?
> 
> Sure, you’re welcome to send a patch for this, or I can change it later today. Thanks!

Sure thing! I can open a pr for that shortly.

**Comment by a8m** [comment]
Closed with #1098
